### PR TITLE
BaseHtml fix renderSelectOptions for Enum

### PR DESCRIPTION
**Pull request by ionutwho**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  BaseHtml fix renderSelectOptions for Enum |

### CODE REVIEW

**Review by bizley — CHANGES_REQUESTED** [review]

File: framework/helpers/BaseHtml.php [hunk 1]
```diff
@@ -1935,7 +1935,7 @@ public static function renderSelectOptions($selection, $items, &$tagOptions = []
                         } elseif ($strict) {
                             $selected = !strcmp((string)$key, (string)$selection);
                         } else {
-                            $selected = $selection == $key;
+                            $selected = is_object($selection) ? $selection->value == $key : $selection == $key;
```
[the remark this answers is not in the document]
Comment (ionutwho) [reply framework/helpers/BaseHtml.php:1938]: Do you mean to check if its a Instance of Enum?

[the remark this answers is not in the document]
Comment (ionutwho) [reply framework/helpers/BaseHtml.php]: added extra check for object: check if it has value property

File: framework/helpers/BaseHtml.php [hunk 2]
```diff
@@ -1935,7 +1935,8 @@ public static function renderSelectOptions($selection, $items, &$tagOptions = []
                         } elseif ($strict) {
                             $selected = !strcmp((string)$key, (string)$selection);
                         } else {
-                            $selected = is_object($selection) ? $selection->value == $key : $selection == $key;
+                            $selected = is_object($selection) && property_exists($selection, 'value') ?
```
Comment (bizley) [inline framework/helpers/BaseHtml.php:1938]: instanceof would be better

**Review by cristiangirlea — COMMENTED** [review]
Pull Request: Improve Selection Matching Logic in renderSelectOptions
Description:
This PR refines the logic for determining if an option is selected in the renderSelectOptions function. The changes ensure clearer and more concise comparisons, improving code readability.

Changes:
Introduced an additional condition to handle cases where $selection is an object with a value property.
Simplified the comparison logic for when $strict is false.

**Review by cristiangirlea — CHANGES_REQUESTED** [review]
Pull Request: Improve Selection Matching Logic in renderSelectOptions
Description:
This PR refines the logic for determining if an option is selected in the renderSelectOptions function. The changes ensure clearer and more concise comparisons, improving code readability.

Changes:
Introduced an additional condition to handle cases where $selection is an object with a value property.
Simplified the comparison logic for when $strict is false.

File: framework/helpers/BaseHtml.php (hunk 2 above)
Comment (cristiangirlea) [reply framework/helpers/BaseHtml.php]: @bizley  Can you check the latest changes

**Comment by ionutwho** [comment]
> I also think that using `instanceof UnitEnum` would be better - it should be safer than changing behavior for all objects.
> 
> And IMO it should be done at the beginning of `$selection` processing:
> 
> ```
> if ($selection !== null) {
>     if ($selection instanceof \UnitEnum) {
>         $selection = $selection->value;
>     }
>     ...
> ```

Well i tried your fix but its not working because we are using this: https://github.com/spatie/enum 
Its not really an Enum, actually its a Class extending an Enum class which doesnt implement UnitEnum...
`abstract class Enum extends \Spatie\Enum\Enum`
`abstract class Enum implements JsonSerializable
`

**Comment by bizley** [comment]
I don't think we should support any object here, this should be out of the scope of the framework and it should depend on the developers to extend framework functionality in their own app to support any custom solution like in your case. Still, we would accept real enums here.

**Comment by Webkadabra** [comment]
are you submitting fixes for some of your internal code to the public? Why are you not checking the type of `value` property then? 
Another useless PR. @samdark I hope we're not thinking of adapting this

**Comment by terabytesoftw** [comment]
Is this PR ready?

**Comment by bizley** [comment]
I'm closing it then. Still, PR introducing support for enums is welcome.

**Pull request closed without merging**